src/main/resources/templates/misc/compare.html

```diff
@@ -1,236 +1,254 @@
+                var color1 = document.getElementById('color-box1').value;
+                var color2 = document.getElementById('color-box2').value;
+
+                if (!file1 || !file2) {
+                  console.error("Please select two PDF files to compare");
+                  return;
+                }
 
-                result2.addEventListener('scroll', function() {
-                  result1.scrollTop = result2.scrollTop;
-                });
+                pdfjsLib.GlobalWorkerOptions.workerSrc = './pdfjs-legacy/pdf.worker.mjs';
 
-                async function comparePDFs() {
-                  const file1 = document.getElementById("fileInput-input").files[0];
-                  const file2 = document.getElementById("fileInput2-input").files[0];
-                  var color1 = document.getElementById('color-box1').value;
-                  var color2 = document.getElementById('color-box2').value;
+                const [pdf1, pdf2] = await Promise.all([
+                  pdfjsLib.getDocument(URL.createObjectURL(file1)).promise,
+                  pdfjsLib.getDocument(URL.createObjectURL(file2)).promise
+                ]);
 
-                  if (!file1 || !file2) {
-                    console.error("Please select two PDF files to compare");
-                    return;
+                const extractText = async (pdf) => {
+                  const pages = [];
+                  for (let i = 1; i <= pdf.numPages; i++) {
+                    const page = await pdf.getPage(i);
+                    const content = await page.getTextContent();
+                    const strings = content.items.map(item => item.str);
+                    pages.push(strings.join(" "));
                   }
-                  pdfjsLib.GlobalWorkerOptions.workerSrc = './pdfjs-legacy/pdf.worker.mjs'
-                  const [pdf1, pdf2] = await Promise.all([
-                    pdfjsLib.getDocument(URL.createObjectURL(file1)).promise,
-                    pdfjsLib.getDocument(URL.createObjectURL(file2)).promise
-                  ]);
-
-                  const extractText = async (pdf) => {
-                    const pages = [];
-                    for (let i = 1; i <= pdf.numPages; i++) {
-                      const page = await pdf.getPage(i);
-                      const content = await page.getTextContent();
-                      const strings = content.items.map(item => item.str);
-                      pages.push(strings.join(" "));
-                    }
-                    return pages.join(" ");
-                  };
+                  return pages.join(" ");
+                };
+
+                const [text1, text2] = await Promise.all([
+                  extractText(pdf1),
+                  extractText(pdf2)
+                ]);
+
+                if (text1.trim() === "" || text2.trim() === "") {
+                  alert("One or both of the selected PDFs have no text content. Please choose PDFs with text for comparison.");
```

Comment:
Know its unrelated but can you move this to the messages files as well since its touching it